src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf

```diff
@@ -423,23 +423,24 @@ def processProcessArgs(Map args) {
   def processArgs = thisDefaultProcessArgs + args
 
   // check whether 'key' exists
-  assert processArgs.containsKey("key")
+  assert processArgs.containsKey("key") : "Error in module '${thisConfig.functionality.name}': key is a required argument"
 
   // if 'key' is a closure, apply it to the original key
   if (processArgs["key"] instanceof Closure) {
     processArgs["key"] = processArgs["key"](thisConfig.functionality.name)
   }
-  assert processArgs["key"] instanceof CharSequence
-  assert processArgs["key"] ==~ /^[a-zA-Z_][a-zA-Z0-9_]*$/
+  def key = processArgs["key"]
+  assert key instanceof CharSequence : "Expected process argument 'key' to be a String. Found: class ${key.getClass()}"
+  assert key ==~ /^[a-zA-Z_][a-zA-Z0-9_]*$/ : "Error in module '$key': Expected process argument 'key' to consist of only letters, digits or underscores. Found: ${key}"
 
   // check whether directives exists and apply defaults
-  assert processArgs.containsKey("directives")
-  assert processArgs["directives"] instanceof Map
+  assert processArgs.containsKey("directives") : "Error in module '$key': directives is a required argument"
```

Comment:
I'm not convinced. The result would be something like:

```groovy
  assert processArgs.containsKey("directives") : "Error in module '$key': directives is a required argument"

  // ↓

  assertMapContainsKey(processArgs, "directives", key)
```

I think it's actually less readable because now the user needs to know what `assertMapContainsKey` actually does.

Reply:
This not true for all functions, no? I think `assertMapContainsKey` explains it pretty well. Could also make the signature a closure with just one argument by already filling in `key`?